dual_net.py

```diff
@@ -0,0 +1,155 @@
+import tensorflow as tf
+import tensorflow.contrib.layers as layers
+
+
+# A convolutional block as described in AlphaGo Zero
+def conv_block(tensor, specs):
+    tensor = layers.convolution2d(tensor,
+                                  num_outputs=specs['num_outputs'],
+                                  kernel_size=specs['kernel_size'],
+                                  stride=specs['stride'],
+                                  activation_fn=None)
+    tensor = tf.layers.batch_normalization(tensor)
+    tensor = tf.nn.relu(tensor)
+    return tensor
+
+
+# A residual block as described in AlphaGo Zero
+def residual_block(tensor, specs):
+    input_tensor = tensor
+    tensor = conv_block(tensor, specs)
+    tensor = layers.convolution2d(tensor,
+                                  num_outputs=specs['num_outputs'],
+                                  kernel_size=specs['kernel_size'],
+                                  stride=specs['stride'],
+                                  activation_fn=None)
+    tensor = tf.layers.batch_normalization(tensor)
+    tensor += input_tensor
```

Comment:
So this step will only work if stride=1? Otherwise the output will be a different size than the input?